x/evidence/keeper/abci.go

```diff
@@ -18,13 +18,13 @@ func (k Keeper) BeginBlocker(ctx context.Context) error {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
 
 	bi := k.cometInfo.GetCometBlockInfo(ctx)
-	if bi == nil {
+	if bi.Evidence == nil {
```

Comment:
this keeps the interface one layer down, should we remove this as well and do an array of evidence?